sycl/include/sycl/nd_item.hpp

```diff
@@ -384,12 +384,12 @@ template <int Dimensions = 1> class nd_item {
     using QualSrcT =
         std::conditional_t<std::is_const_v<SrcT>, const uint8_t, uint8_t>;
     auto DestP = multi_ptr<uint8_t, DestS, access::decorated::yes>(
-        detail::cast_AS<typename multi_ptr<uint8_t, DestS,
-                                           access::decorated::yes>::pointer>(
+        reinterpret_cast<typename multi_ptr<uint8_t, DestS,
```

Comment:
We're changing element type of the decorated C++ pointer type here, no address space changes. Same throughout this file.